invenio_records_files/views.py

```diff
@@ -25,68 +29,128 @@ def create_blueprint_from_app(app):
-        __name__,
-        url_prefix='')
-
-    for config_name, endpoints_to_register in \
-            iteritems(app.config['RECORDS_FILES_REST_ENDPOINTS']):
-        for endpoint_to_register in endpoints_to_register:
-            record_item_path = \
-                app.config[config_name][endpoint_to_register]['item_route']
-            files_resource_endpoint_suffix = \
-                endpoints_to_register[endpoint_to_register]
-            files_resource_endpoint_suffix = \
-                urljoin('/', files_resource_endpoint_suffix)
+        "invenio_records_files", __name__, url_prefix=""
+    )
+
+    for config_name, endpoints_to_register in iteritems(
+        app.config["RECORDS_FILES_REST_ENDPOINTS"]
+    ):
+        for endpoint_prefix in endpoints_to_register:
+            record_item_path = app.config[config_name][endpoint_prefix][
+                "item_route"
+            ]
+            files_resource_path = endpoints_to_register[endpoint_prefix]
+            files_resource_path = urljoin("/", files_resource_path)
+            bucket_view = RecordBucketResource.as_view(
+                endpoint_prefix + "_bucket_api",
+                serializers={
+                    "application/json": partial(
+                        json_serializer,
+                        view_name="{}_bucket_api".format(endpoint_prefix),
+                        serializer_mapping=serializer_mapping,
+                    )
+                },
+            )
+            object_view = RecordObjectResource.as_view(
+                endpoint_prefix + "_object_api",
+                serializers={
+                    "application/json": partial(
+                        json_serializer,
+                        view_name="{}_object_api".format(endpoint_prefix),
+                        serializer_mapping=serializer_mapping,
+                    )
+                },
+            )
             records_files_blueprint.add_url_rule(
-                '{record_item_path}{files_resource_endpoint_suffix}'
-                .format(**locals()),
+                "{record_item_path}{files_resource_path}".format(**locals()),
                 view_func=bucket_view,
             )
             records_files_blueprint.add_url_rule(
-                '{record_item_path}{files_resource_endpoint_suffix}/<path:key>'
-                .format(**locals()),
+                "{record_item_path}{files_resource_path}/<path:key>".format(
+                    **locals()
+                ),
                 view_func=object_view,
             )
 
-    @records_files_blueprint.url_value_preprocessor
-    def resolve_pid_to_bucket_id(endpoint, values):
-        """Flask URL preprocessor to resolve pid to Bucket ID.
+    return records_files_blueprint
+
+
+def pass_bucket_id(f):
+    """Decorate to retrieve a bucket."""
+
+    @wraps(f)
+    def decorate(*args, **kwargs):
+        # We need to make sure to pass an empty string and not None which can
```

Comment:
I would change this comment for something like ( it took me a while to understand it maybe because of the sentence structure)
``` 
# We need to make sure to pass empty string instead of None to make sure it will not be set as a property value of the Record,
# because None property value is "reserved" for Record with files but no attached bucket
```